modules/Bio/EnsEMBL/Compara/RunnableDB/GenomeDBFactory.pm

```diff
@@ -121,6 +123,21 @@ sub fetch_input {
     $genome_dbs = [grep {not $_->genome_component} @$genome_dbs] if not $self->param('component_genomes');
     $genome_dbs = [grep {($_->name eq 'ancestral_sequences') or $_->is_polyploid or $_->genome_component} @$genome_dbs] if not $self->param('normal_genomes');
 
+
+    # filter out exclude_species
+    my %exclude_species;
+    my $_exclude_species = $self->param('exclude_species');
```

Comment:
Following the general code style of compara, we never precede a variable name with `_`. A better way to distinguish might be to name the parameter variable `$exclude_species`, and the array version later on something like `$excluded_species`. The same goes for `@_gdbs`. In Perl, the preceding underscore convention is restricted to subroutines and variables locally scoped in just those subrutines.